### PR TITLE
Add CliTool project and implement general commands

### PR DESCRIPTION
**Pull request by yamaoto**
Add CliTool #5 

Usage:
```
> cd Cynosura.Studio.CliTool
> dotnet pack
> dotnet tool install --global --add-source ./nupkg Cynosura.Studio.CliTool
> cyn help
```

### CODE REVIEW

**Comment by yamaoto** [comment]
What is the status of the PR?

File: Cynosura.Studio.CliTool/Commands/InfoCommand.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            : base(solutionDirectory, feed, src, templateName, lifetimeScope)
+        {
+        }
+
+        public override Task<bool> ExecuteAsync(string[] args)
+        {
+            var accessor = new SolutionAccessor(SolutionDirectory);
+            var meta = accessor.Metadata;
+            Console.WriteLine(
+                $"Solution name: {meta.Name}\r\nTemplate name:{meta.TemplateName}\r\nTemplate version: {meta.TemplateVersion}");
```
Comment (bacr) [inline Cynosura.Studio.CliTool/Commands/InfoCommand.cs]: Missing space after "Template name"

File: Cynosura.Studio.CliTool/Commands/GenerateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                throw new Exception($"Entity {name} not found");
+            }
+            await generator.GenerateEntityAsync(accessor, entity);
+            await generator.GenerateViewAsync(accessor, new View(), entity);
+        }
+
+        public override string Help()
+        {
+            return $"{CliApp.CommandName} generate <action>\r\n" +
+                   $"Available actions: \r\n{string.Join("\r\n\t", _actions.Keys)}\r\n";
```
Comment (bacr) [inline Cynosura.Studio.CliTool/Commands/GenerateCommand.cs]: Please align help messages accordingly. Especially when showing lists, sometimes first element is on the same line, and sometimes it's on the next line, but is not indented like other elements.

File: Cynosura.Studio.CliTool/Cynosura.Studio.CliTool.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="2.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="2.1.1" />
+    <PackageReference Include="Autofac" Version="4.8.1" />
+    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.3.0" />
+    <PackageReference Include="ConsoleTables" Version="2.2.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Cynosura.Studio.Core\Cynosura.Studio.Core.csproj" />
```
Comment (bacr) [inline Cynosura.Studio.CliTool/Cynosura.Studio.CliTool.csproj]: Change reference to Studio.Generator

**Review by bacr — CHANGES_REQUESTED** [review]

File: Cynosura.Studio.CliTool/Commands/InfoCommand.cs (hunk 1 above)
Comment (yamaoto) [reply Cynosura.Studio.CliTool/Commands/InfoCommand.cs]: Solved

File: Cynosura.Studio.CliTool/Commands/GenerateCommand.cs (hunk 2 above)
Comment (yamaoto) [reply Cynosura.Studio.CliTool/Commands/GenerateCommand.cs]: Solved

File: Cynosura.Studio.CliTool/Cynosura.Studio.CliTool.csproj (hunk 3 above)
Comment (yamaoto) [reply Cynosura.Studio.CliTool/Cynosura.Studio.CliTool.csproj]: Solved

**Review by bacr — APPROVED** [review]

**Pull request merged**